pandas/tests/reshape/test_melt.py

```diff
@@ -661,3 +661,36 @@ def test_col_substring_of_stubname(self):
                                  i=['node_id', 'A'],
                                  j='time')
         tm.assert_frame_equal(result, expected)
+
+    def test_melt_missing_columns(self):
+        # GH-23575
+        # This test is to ensure that pandas raises an error if melting is
+        # attempted with column names absent from the dataframe
+
+        # Generate data
+        df = pd.DataFrame(np.random.randn(5, 4), columns=list('abcd'))
+
+        # Try to melt with missing `value_vars` column name
+        with pytest.raises(KeyError, match="The following 'value_vars' are not"
+                                           " present in"
+                                           " the DataFrame: 'C'"):
+            df.melt(['a', 'b'], ['C', 'd'])
+
+        # Try to melt with missing `id_vars` column name
+        with pytest.raises(KeyError, match="The following 'id_vars' are not"
+                                           " present in"
+                                           " the DataFrame: 'A'"):
+            df.melt(['A', 'b'], ['c', 'd'])
+
+        # Try with error in both-> `id_vars` caught first
```

Comment:
I think these last two are relatively less useful. If you wanted, you could check both for missing `id_vars` and `value_vars` before raising, so that the error message contains *all* the issues. Feel free to implement that I you want, but I don't have a strong preference.